quinn-proto/src/endpoint.rs

```diff
@@ -109,7 +109,15 @@ impl Endpoint {
                 if let EndpointEvent::Incoming = &event {
                     self.incoming.push_back(ch);
                 }
+                let done = if let EndpointEvent::Closed { .. } = &event {
+                    true
+                } else {
+                    false
+                };
                 self.handle_event(ch, event);
+                if done {
+                    break;
+                }
```

Comment:
Is this necessary? Are there cases where extra endpoint events might trail after `Closed`?

Reply:
With how the loop is setup, there's always at least one `poll_endpoint_events()` for the next loop iteration which we have to prevent, right?

Reply:
If that call will return `None`, why do we need to prevent it?

Reply:
Because it indexes into `self.connections[ch]` with a `ConnectionHandle` which is no longer present there?

Reply:
Ah, of course.